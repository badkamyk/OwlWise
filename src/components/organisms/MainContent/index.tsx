import { animated, useTrail } from "@react-spring/web";
import { useInView } from "react-intersection-observer";
import MainContentImages from "../../molecules/MainContentImages";
import MainContentText from "../../molecules/MainContentText";

const textInfo = {
	headingText: "We didn't reinvent the wheel",
	paragraphText:
		"We are strategists, designers and developers. Innovators and problem solvers. Small enough " +
		"to be simple and quick, but big enough to deliver the scope you want at the pace you need. " +
		"Small enough to be simple and quick, but big enough to deliver the scope you want at the pace you need.",
	secondaryText:
		"We act as an extension of your team, working with you to deliver the best possible product.",
	headingClassName:
		"mb-4 text-4xl md: text-5xl tracking-tight font-extrabold text-gray-900",
	paragraphClassName: "mb-4 text-gray-900 text-md md:text-2xl",
};

export default function MainContent() {
	const { ref, inView } = useInView({
		threshold: 0.5,
		triggerOnce: true,
	});

	const trail = useTrail(2, {
		opacity: inView ? 1 : 0,
		transform: inView ? "translate3d(0,0px,0)" : "translate3d(0,-40px,0)",
		config: { duration: 900 },
	});

	return (
		<section
			className="divide-y-8 bg-gradient-to-b from-teal-100 via-rose-100 to-gray-500 py-12 text-center"
			data-testid="main-content"
		>
			<div
				className="mx-auto max-w-screen-xl items-center gap-16 py-8 px-4 lg:grid lg:grid-cols-2 lg:py-16 lg:px-6"
				ref={ref}
			>
				{trail.map(({ ...style }, index) => (
					<animated.div key={index + "mainContent"} style={style}>
						{index === 0 ? (
							<MainContentText
								headingText={textInfo.headingText}
								paragraphText={textInfo.paragraphText}
								secondaryText={textInfo.secondaryText}
							/>
						) : (
							<MainContentImages
								firstSrc={
									"https://flowbite.s3.amazonaws.com/blocks/marketing-ui/content/office-long-2.png"
								}
								firstAlt={"People in the office"}
								secondSrc={
									"https://flowbite.s3.amazonaws.com/blocks/marketing-ui/content/office-long-1.png"
								}
								secondAlt={"Office equipment"}
							/>
						)}
					</animated.div>
				))}
			</div>
		</section>
	);
}
