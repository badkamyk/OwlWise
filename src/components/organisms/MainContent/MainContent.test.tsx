import { render, screen } from "@testing-library/react";
import "intersection-observer";
import { describe, expect, it } from "vitest";
import MainContent from "./index";

describe("MainContent", () => {
	it("renders MainContent correctly with Heading, Features and Footer", () => {
		const { container } = render(<MainContent />);
		expect(screen.getByTestId("main-content")).toBeTruthy();
		expect(screen.getByTestId("main-content-text")).toBeTruthy();
		expect(screen.getByTestId("main-content-images")).toBeTruthy();
		expect(container).toMatchSnapshot();
	});
});
