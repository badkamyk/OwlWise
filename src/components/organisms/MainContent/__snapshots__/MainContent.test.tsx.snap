// Vitest Snapshot v1

exports[`MainContent > renders MainContent correctly with Heading, Features and Footer 1`] = `
<div>
  <section
    class="divide-y-8 bg-gradient-to-b from-teal-100 via-rose-100 to-gray-500 py-12 text-center"
    data-testid="main-content"
  >
    <div
      class="mx-auto max-w-screen-xl items-center gap-16 py-8 px-4 lg:grid lg:grid-cols-2 lg:py-16 lg:px-6"
    >
      <div
        style="opacity: 0; transform: translate3d(0,-40px,0);"
      >
        <div
          class="font-light text-gray-500 dark:text-gray-400 sm:text-lg"
          data-testid="main-content-text"
        >
          <h3
            class="mb-4 tracking-tight font-extrabold text-gray-900 text-4xl"
          >
            We didn't reinvent the wheel
          </h3>
          <div
            class="font-normal"
          >
            <p
              class="mb-4 text-gray-900 text-xl"
              data-testid="main-content-paragraph"
            >
              We are strategists, designers and developers. Innovators and problem solvers. Small enough to be simple and quick, but big enough to deliver the scope you want at the pace you need. Small enough to be simple and quick, but big enough to deliver the scope you want at the pace you need.
            </p>
            <p
              class="mb-4 text-gray-900 text-xl"
              data-testid="main-content-paragraph"
            >
              We act as an extension of your team, working with you to deliver the best possible product.
            </p>
          </div>
        </div>
      </div>
      <div
        style="opacity: 0; transform: translate3d(0,-40px,0);"
      >
        <div
          class="mt-8 grid grid-cols-2 gap-4"
          data-testid="main-content-images"
        >
          <img
            alt="People in the office"
            src="https://flowbite.s3.amazonaws.com/blocks/marketing-ui/content/office-long-2.png"
          />
          <img
            alt="Office equipment"
            src="https://flowbite.s3.amazonaws.com/blocks/marketing-ui/content/office-long-1.png"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
